pkg/discovery/discovery.go

```diff
@@ -80,6 +81,17 @@ func Run(kubeClient kubernetes.Interface, cfg *config.Config) []error {
 		errlst = append(errlst, err)
 	}
 
+	// 8. Always write the done file
+	doneFile := cfg.ResultsDir + "/done"
```

Comment:
This resulted in the least amount of changes to the CLI. If we make the assumption we always want to write a done file (which we can choose to react to or not) then this becomes the easiest way forward.

It works well with the `--no-exit flag`. This change allows us to get both the blocking behavior as well as a `done` file. It's not trivial to get this behavior by only adding a script. If sonobuoy never exits, we'd need a second process watching sonobuoy waiting for it to get to the blocking state and *then* write a done file.